src/components/recordedit/form-container.tsx

```diff
@@ -216,74 +335,204 @@ const FormRow = ({ columnModelIndex }: FormRowProps): JSX.Element => {
+              // I couldn’t test that scenario, since on load we’re removing the forms that user cannot edit,
+              if (
+                appMode === appModes.EDIT &&
+                canUpdateValues &&
+                !canUpdateValues[`${formNumber}-${cm.column.name}`]
+              ) {
+                return;
+              }
+              handleFormClick(formNumber);
+            }}
+          >
             {renderInput(formNumber, formIndex)}
           </div>
         ))}
       </div>
-      {showSelectAll && <SelectAllRow columnModelIndex={columnModelIndex} />}
+      {showSelectAll && (
+        <SelectAllRow
+          activeForm={activeForm}
+          setActiveForm={setActiveForm}
+          columnModelIndex={columnModelIndex}
+        />
+      )}
     </div>
-  )
-
+  );
 };
 
 /**
  * shows the select all row
  * NOTE this is its own component to avoid rerendering the whole row on each change.
  */
-const SelectAllRow = ({ columnModelIndex }: FormRowProps) => {
+const SelectAllRow = ({
+  columnModelIndex,
+  activeForm,
+  setActiveForm,
+}: FormRowProps) => {
   const {
-    columnModels, forms, reference, waitingForForeignKeyData, foreignKeyData, appMode,
-    canUpdateValues, toggleActiveSelectAll, logRecordeditClientAction
+    columnModels,
+    forms,
+    reference,
+    waitingForForeignKeyData,
+    foreignKeyData,
+    appMode,
+    canUpdateValues,
+    toggleActiveSelectAll,
+    logRecordeditClientAction,
   } = useRecordedit();
+  const {
+    watch,
+    reset,
+    getValues,
+    formState: { errors },
+  } = useFormContext();
+  let ref = useRef(null);
+  const [isEmpty, setIsEmpty] = useState(true);
+  const [selected, setSelected] = useState(false);
+  const [showTooltip, setShowTooltip] = useState(false);
+  const [width, setWidth] = useState(0);
 
-  const { watch, reset, getValues, formState: { errors } } = useFormContext();
+  // This useeffect is to set the first form as active on load
+  useEffect(() => {
+    if (activeForm?.length === 0) {
+      setActiveForm([forms[0]]);
+    }
+  }, []);
 
-  const [isEmpty, setIsEmpty] = useState(true);
+  /* This useffect is to set the indeterminate checkbox when forms are individually selected
+   *  and selected forms length is less than total forms length
+   */
+  useEffect(() => {
+    if (ref && ref.current && activeForm) {
+      (ref.current as HTMLInputElement).indeterminate =
+        activeForm?.length > 0
+          ? activeForm?.length < forms.length
+            ? true
+            : false
+          : false;
+      setSelected(activeForm?.length === forms.length);
+    }
+  }, [activeForm]);
+
+  // useEffect to call update only when there is a change in the width of select-all-row to update textarea width
+  useEffect(() => {
+    updateTextareaWidth();
+  }, [width]);
+
+  // useEffect to have a resize sensor to change the direction of upper-row when width of container is less than 400.
+  useEffect(() => {
```

Comment:
You're just explaining what this does and not why. You should mention you're doing this to make sure buttons are on top of the rest. You also need to explain why 400. Why not 350 or any other values? Is it calculated based on the min-width of textarea or something like that?

I also expected this to work for normal inputs, but it's only working in textarea case.